### PR TITLE
Fixes Issue #357 - ESXi DHCP Model

### PR DESCRIPTION
**Pull request by jcpowermac**
This PR resolves issue #357.  

Testing comments will follow.

### CODE REVIEW

**Comment by jcpowermac** [comment]
Testing results:

Model using new vmware_esx_5_dhcp type.
```
root@6e65c62015f6:/home/hanlon# cli/hanlon model 6D2
Model:
 Label =>  vmware_dhcp
 Template =>  vmware_hypervisor
 Description =>  vmware esxi 5_dhcp
 UUID =>  6D2yjqamVu0sk17xGZuHfX
 Image UUID =>  2bViaT307xsqgopw2tBmYl
```

```
root@6e65c62015f6:/home/hanlon# cli/hanlon active_model PjP logs  

             State                   Action             Result          Time      Last     Total            Node           
init                             mk_call          n/a                 17:49:22  0 sec     0 sec     MFB2ZrZKtL0EKJDIdSzzv  
init                             boot_call        n/a                 17:49:45  23 sec    23 sec    MFB2ZrZKtL0EKJDIdSzzv  
init                             boot_call        n/a                 17:53:19  3.6 min   4.0 min   MFB2ZrZKtL0EKJDIdSzzv  
init                             kickstart_file   n/a                 17:55:23  2.1 min   6.0 min   MFB2ZrZKtL0EKJDIdSzzv  
init=>postinstall                kickstart_end    n/a                 17:55:23  0 sec     6.0 min   MFB2ZrZKtL0EKJDIdSzzv  
postinstall                      boot_call        n/a                 18:05:44  10.3 min  16.4 min  MFB2ZrZKtL0EKJDIdSzzv  
postinstall=>os_complete         postinstall_end  n/a                 18:13:00  7.3 min   23.6 min  MFB2ZrZKtL0EKJDIdSzzv  
os_complete=>complete_no_broker  broker_check     No broker attached  18:13:00  0 sec     23.6 min  MFB2ZrZKtL0EKJDIdSzzv  
```

The included `@node_ip`
```
[root@localhost Hanlon]# curl -g "http://10.53.252.254:8026/hanlon/api/v1/active_model/PjP3Fag2VRq3HKio7wwyp" | jq .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 12821  100 12821    0     0   133k      0 --:--:-- --:--:-- --:--:--  134k
{
  "response": {
    "@root_policy": "7XnZTfCZiNisjdwOp2VeVP",
    "@broker": null,
    "@model": {
      "@node_ip": "10.53.252.243",
```

**Comment by mtnbikenc** [comment]
Running another test this morning (including the new commit) to see how it goes.

File: core/model/esxi/5_dhcp/kickstart.erb [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+def enable_firewall():
+    output, err, return_code = execute_process("esxcli network firewall set --enabled yes")
+
+
+def disable_firewall():
+    output, err, return_code = execute_process("esxcli network firewall set --enabled no")
+
+
+disable_firewall()
+send_node_ip_callback()
```
Comment (mtnbikenc) [inline core/model/esxi/5_dhcp/kickstart.erb]: As discussed with @jcpowermac, the timing of operations on the host is causing the firewall to not quite be down before issuing the send_ips command.  Adding a sleep between these tasks solved that issue.  So, we need to have better handling of the completion and execution of these steps.

modified kickstart.erb
```
disable_firewall()
time.sleep(10)            # added sleep here
send_node_ip_callback()
send_postinstall_end()
enable_firewall()
```

**Comment by tjmcs** [comment]
still testing this PR, @mtnbikenc?

**Comment by mtnbikenc** [comment]
With the last commit, looks good to merge.  :+1:

**Pull request merged**